sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -39,12 +44,31 @@ class RapidsGdsStore(
         case d: DeviceMemoryBuffer => d
         case _ => throw new IllegalStateException("copying from buffer without device memory")
       }
-      val id = other.id
-      val path = if (id.canShareDiskPaths) {
-        sharedBufferFiles.computeIfAbsent(id, _ => id.getDiskPath(diskBlockManager))
+      if (deviceBuffer.getLength < batchWriteBufferSize) {
+        batchSpiller.spill(other, deviceBuffer)
       } else {
-        id.getDiskPath(diskBlockManager)
+        singleShotSpiller.spill(other, deviceBuffer)
       }
+    }
+  }
+
+  abstract class RapidsGdsBuffer(
+      override val id: RapidsBufferId,
+      override val size: Long,
+      override val meta: TableMeta,
+      spillPriority: Long,
+      override val spillCallback: RapidsBuffer.SpillCallback)
+      extends RapidsBufferBase(id, size, meta, spillPriority, spillCallback) {
+    override val storageTier: StorageTier = StorageTier.GDS
+
+    override def getMemoryBuffer: MemoryBuffer = getDeviceMemoryBuffer
+  }
+
+  class SingleShotSpiller() {
```

Comment:
I like seeing the batch logic encapsulated in a separate class to keep the main GDS store cleaner, but in this single-shot case it doesn't seem worth it. This class has no instance data, implements no common interface, and provides only one method.  This makes me wonder why not move `RapidsGdsSingleShotBuffer` out of this class and just make the `spill` method here named `singleShotSpill` in RapidsGdsStore directly.  I'm fine if we keep it, but it should be an `object` rather than a `class` since the method can be static.